artemis-core/artemis/src/main/java/com/artemis/ComponentClassProvider.java

```diff
@@ -0,0 +1,15 @@
+package com.artemis;
+
+import com.artemis.annotations.UnstableApi;
+
+import java.util.Collection;
+
+/**
+ * Class provider used for Aspects
+ *
+ * @author Felix Bridault
+ */
+@UnstableApi
+public interface ComponentClassProvider {
+    Collection<Class<? extends  Component>> getComponents();
```

Comment:
maybe change `Collection` to `Iterable` (or `Iterator` - if `Iterable` breaks existing code)?  This would change existing all/one/exclude in the Builder too.

Reply:
How about just making ComponentClassProvider Iterable itself?